disdrodb/L0/L0_processing.py

```diff
@@ -443,3 +447,162 @@ def df_sanitizer_fun(df, lazy=False)
+    -------
+    bool
+        True if the data source exists.
+        False if it does not exist.
+
+    Raises
+    ------
+    ValueError
+        Error if the data source name provided has not been found.
+    """
+
+    dict_all_readers = get_available_readers()
+
+    if dict_all_readers.get(data_source, None):
+        data_source_exists = True
+    else:
+        data_source_exists = False
+        msg = (
+            f"Data source {data_source} has not been found within the available readers"
+        )
+        logger.exception(msg)
+        raise ValueError(msg)
+
+    return data_source_exists
+
+
+def get_available_readers_by_data_source(data_source: str) -> dict:
+    """Return the available readers by data source
+
+    Parameters
+    ----------
+    data_source : str
+        Data source name - Institution name (when campaign data spans more than 1 country) or country (when all campaigns (or sensor networks) are inside a given country)
+
+    Returns
+    -------
+    dict
+        Dictionary that conatins the campaigns for the requested data source.
+
+    """
+
+    if check_data_source(data_source):
+        dict_data_source = get_available_readers.get(data_source, None)
+
+    return dict_data_source
+
+
+def check_reader_name(data_source: str, reader_name: str) -> bool:
+    """Check if the provided data source exists and reader names exists within the available readers.
+    Please run get_available_readers() to get the list of all available reader
+
+    Parameters
+    ----------
+    data_source : str
+        Data source name - Institution name (when campaign data spans more than 1 country) or country (when all campaigns (or sensor networks) are inside a given country)
+    reader_name : str
+        Campaign name
+
+    Returns
+    -------
+    bool
+        True if the reader has been found,
+        False if the reader has not been found.
+
+    Raises
+    ------
+    ValueError
+        Error if the reader name provided for the campaign has not been found.
+    """
+
+    if check_data_source(data_source):
+        if get_available_readers().get(data_source).get(reader_name, None):
+            reader_exist = True
+        else:
+            reader_exist = False
+            msg = (
+                f"Reader {reader_name} has not been found within the available readers"
+            )
+            logger.exception(msg)
+            raise ValueError(msg)
+    return reader_exist
+
+
+def get_reader(data_source: str, reader_name: str) -> object:
+    """Returns the reader function based on input parameters
+
+    Parameters
+    ----------
+    data_source : str
+        Institution name (when campaign data spans more than 1 country) or country (when all campaigns (or sensor networks) are inside a given country)
+    reader_name : str
+        Campaign name
+
+    Returns
+    -------
+    object
+        The reader() function
+
+    """
+
+    reader_exist = check_reader_name(data_source, reader_name)
```

Comment:
I would return from `check_reader_name` the valid `reader_name`.
Then the if conditions is not required anymore (already now it does not make sense ...)